initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+                            ${tier.pricePerTicket.toFixed(0)} NOK each<br>
+                            <small>(Tickets ${tier.absoluteRange} total)</small>
+                        </div>
+                    `
+                      )
+                      .join("")}
+                </div>
+            </div>
+        `;
+  }
+
+  displayComparison(comparisons, reisCost, period) {
+    const comparisonSection = document.getElementById("comparison");
+
+    if (comparisons.length === 0) {
+      comparisonSection.innerHTML = "";
+      return;
+    }
+
+    const periodText = this.getPeriodText(period);
+
+    comparisonSection.innerHTML = `
+            <h3>💰 Ticket Comparison for ${periodText}</h3>
+            <p>Here's how different ticket options compare for your usage:</p>
+
+            <div class="comparison-grid">
+                ${comparisons
+                  .map(
+                    (option) => `
+                    <div class="ticket-option ${
+                      option.isRecommended ? "recommended" : ""
+                    }">
+                        <h4>${option.name}</h4>
+                        <div class="price-display">${option.totalCost.toFixed(
+                          0
+                        )} NOK</div>
+                        <div class="price-per-trip">${option.costPerTrip.toFixed(
+                          1
+                        )} NOK per trip</div>
+                        ${
+                          option.savings > 0
+                            ? `<div class="savings-badge">Save ${option.savings.toFixed(
+                                0
+                              )} NOK</div>`
+                            : option.savings < 0
+                            ? `<div style="color: #dc3545; font-size: 0.9rem;">+${Math.abs(
+                                option.savings
+                              ).toFixed(0)} NOK more</div>`
+                            : ""
+                        }
+                        ${
+                          option.type !== "reis" && option.ticketsNeeded > 1
+                            ? `<div style="color: #666; font-size: 0.8rem; margin-top: 0.5rem;">${option.ticketsNeeded} tickets needed</div>`
+                            : ""
+                        }
+                    </div>
+                `
+                  )
+                  .join("")}
+            </div>
+
+            <div style="margin-top: 1.5rem; padding: 1rem; background: #e3f2fd; border-radius: 6px;">
+                <strong>💡 Recommendation:</strong>
+                ${this.getRecommendationText(comparisons)}
+            </div>
+        `;
+  }
+
+  getZoneText(zone) {
+    const zoneMap = {
+      1: "Zone 1 (Oslo city center)",
+      2: "Zones 1-2 (Oslo + inner suburbs)",
+      3: "Zones 1-3 (Oslo + outer suburbs)",
+      4: "Zones 1-4 (Oslo + extended area)",
+      all: "All zones (Oslo + Akershus)",
+    };
+    return zoneMap[zone] || "Unknown zone";
+  }
```

Comment:
🔧 Move this to a top-level constant, give it a type, and catch incorrect zones at compile time rather than runtime.